explainer.md

```diff
@@ -300,6 +300,18 @@ input.readable.pipeInto(demuxer.writable);
 muxer.readable.pipeInto(output.writable);
 ```
 
+## Detailed design discussion
+
+### Execution environment
+
+Encoder and decoder objects can be instantiated on the main thread and on dedicated workers (TODO: service workers too?). Additionally, the ReadableStream and WritableStream components of each object can be transferred to other contexts (utilizing the [transferrable stream][transferable-streams] infrastructure).
+
+Encode and decode operations can be very computationally expensive. As such, user agents must perform the operations asynchronously with the JavaScript which initiates the operation. The execution environment of the codec implementation is defined by the user agent. Some possibilities include: sequentially or in parallel on an internal thread pool, or on a hardware acceleration unit (e.g., a GPU).
+
+By building on the Streams infrastructure, WebCodecs allows the user agent to optimize the transfer of chunks between JavaScript and internal execution environments.
```

Comment:
Yes, it's critical that this design rational is well understood. Not only for chunks, but only for frames, texture handles, etc. (depending on the platform). Or maybe that's what you meant already? The terminology is still a bit in flux :-).

Reply:
"Chunks" is a term taken from the Streams infrastructure and refers to the data that flows in the stream. I added some clarification text about handling expensive resources.